src/commands/info.ts

```diff
@@ -18,32 +20,69 @@ export const builder: CommandBuilder<{}, Options> = (yargs) =>
   }) as yargs.Argv<Options>;
 
 export const handler = async (argv: Arguments<Options>): Promise<void> => {
+  await init();
   try {
-    let { name } = argv;
-    const [tableId, chainId] = name.split("_").reverse();
+    const { name, aliases, enableEnsExperiment, ensProviderUrl } = argv;
 
-    const parts = name.split("_");
+    let chainId, tableId;
+    // Check if the passed `name` is valid, otherwise, if it's a table alias,
+    // making sure standard table names take precedence
+    try {
+      ({ chainId, tableId } = await globalThis.sqlparser.validateTableName(
+        name
+      ));
+    } catch (err: any) {
+      if (aliases) {
+        try {
+          const nameMap = await jsonFileAliases(aliases).read();
+          const nameFromAlias = nameMap[name];
+          ({ chainId, tableId } = await globalThis.sqlparser.validateTableName(
+            nameFromAlias
+          ));
+        } catch (err: any) {
+          // Throw only if ENS isn't enabled, which is checked next
+          if (!(enableEnsExperiment && ensProviderUrl))
+            logger.error("invalid table alias, table name not found");
+          return;
+        }
+      }
+      // We'll throw later if `chainId` or `tableId` are undefined
+    }
+    // If ENS is enabled, perform a last attempt on validation
+    /* c8 ignore next 11 */
+    if (enableEnsExperiment && ensProviderUrl) {
+      const { ens } = await setupCommand({
+        ...argv,
+      });
+      try {
+        const nameFromEns = await ens?.resolveTable(name);
+        if (nameFromEns)
+          ({ chainId, tableId } = await globalThis.sqlparser.validateTableName(
+            nameFromEns
+          ));
+      } catch (err: any) {
+        logger.error("invalid ENS namespace, table record not found");
+        return;
+      }
+    }
 
-    if (parts.length < 3 && !argv.enableEnsExperiment) {
+    // The "standard" table name was passed and is invalid if these are undefined
+    if (tableId === undefined || chainId === undefined) {
       logger.error(
         "invalid table name (name format is `{prefix}_{chainId}_{tableId}`)"
       );
       return;
     }
 
-    const { ens, validator } = await setupCommand({
+    // Note: should `setupCommand` be used again, or instantiate a validator directly?
+    const { validator } = await setupCommand({
```

Comment:
one question i had re: `setupCommand`—should it only be used once? i.e., i use it above with:
```
const { ens } = await setupCommand({
  ...argv,
});
```
instead, i can instantiate one or the other directly vs using the method.

Reply:
I think this can all be reworked to simply check if an aliases arg was provided with an alias for name, if so do the swap.  Then leave the rest of the logic in this command unchanged.  This would be a convenient spot to use the helper described in the comment on the controller command.

Reply:
i ended up keeping duplicative `setupCommand` calls for when ENS is enabled. the previous logic was using this:
```js
let { name } = argv;
const [tableId, chainId] = name.split("_").reverse();
const parts = name.split("_");
// error if parts are invalid, or if ENS is not enabled
// ...
const { ens, validator } = await setupCommand({
   ...argv,
  chain: parseInt(chainId) as any,
});
```
since an ENS name wont have a `chainId` parsed, using `parseInt(chainId)` won't work properly. thus, i had to use `setupCommand` above to get the table name from ENS, and then get the `chainId` before calling `setupCommand` again for the validator.